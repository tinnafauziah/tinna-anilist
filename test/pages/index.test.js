import flushPromises from 'flush-promises';
import index from '~/pages/index';
import animeMock from '~/test/mocks/store-mocks/anime-mocks';

let store;
const mocks = {
  $router: {
    push: jest.fn(),
  },
};
const modules = { ...animeMock };

describe('Index Page', () => {
  const mockPage = () => {
    store = global.createStore({ ...modules });
    return global.shallowMount(index, { store, mocks });
  };
  it('should index page match snapshot', async () => {
    const wrapper = mockPage();
    await flushPromises();
    expect(wrapper.element).toMatchSnapshot();
  });
});
