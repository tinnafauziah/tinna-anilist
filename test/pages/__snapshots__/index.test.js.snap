// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page should index page match snapshot 1`] = `
<div
  class="index-anime"
>
  <div
    class="display-2 mb-4"
  >
    Search Anime
  </div>
  <v-layout-stub
    tag="div"
    wrap="wrap"
  >
    <v-flex-stub
      class="xs4"
      tag="div"
    >
      <v-select-stub
        appendicon="$dropdown"
        backgroundcolor=""
        clearicon="$clear"
        errorcount="1"
        errormessages=""
        itemcolor="primary"
        itemdisabled="disabled"
        items="Action,Adventure,Comedy,Drama,Ecchi,Fantasy,Hentai,Horror,Mahou Shoujo,Mecha,Music,Mystery,Psychological,Romance,Sci-Fi,Slice of Life,Sports,Supernatural,Thriller"
        itemtext="text"
        itemvalue="value"
        label="Pilih genre"
        loaderheight="2"
        menuprops="[object Object]"
        messages=""
        multiple="true"
        nodatatext="$vuetify.noDataText"
        outlined="true"
        rules=""
        successmessages=""
        type="text"
        value=""
        valuecomparator="[Function]"
      />
    </v-flex-stub>
    <v-flex-stub
      class="xs4 ml-2"
      tag="div"
    >
      <v-select-stub
        appendicon="$dropdown"
        backgroundcolor=""
        clearicon="$clear"
        errorcount="1"
        errormessages=""
        itemcolor="primary"
        itemdisabled="disabled"
        items="[object Object],[object Object],[object Object],[object Object]"
        itemtext="text"
        itemvalue="value"
        label="Sorted by"
        loaderheight="2"
        menuprops="[object Object]"
        messages=""
        nodatatext="$vuetify.noDataText"
        outlined="true"
        rules=""
        successmessages=""
        type="text"
        value="ID"
        valuecomparator="[Function]"
      />
    </v-flex-stub>
    <v-flex-stub
      class="xs1 ml-4 mr-6"
      tag="div"
    >
      <v-btn-stub
        activeclass=""
        color="teal accent-4"
        tag="button"
        type="button"
        xlarge="true"
      >
        Search
      </v-btn-stub>
    </v-flex-stub>
    <v-spacer-stub
      tag="div"
    />
    <v-flex-stub
      class="ml-6"
      tag="div"
    >
      <v-btn-stub
        activeclass=""
        color="teal lighten-4"
        tag="button"
        type="button"
        xlarge="true"
      >
        <v-icon-stub
          dark="dark"
          left="left"
        >
          mdi-bookmark
        </v-icon-stub>
        <span>
          Bookmark Anime List
        </span>
      </v-btn-stub>
    </v-flex-stub>
  </v-layout-stub>
  <v-layout-stub
    class="scroll_container"
    tag="div"
    wrap="wrap"
  >
    <animecard-stub
      anime="[object Object]"
    />
    <animecard-stub
      anime="[object Object]"
    />
    <animecard-stub
      anime="[object Object]"
    />
    <animecard-stub
      anime="[object Object]"
    />
    <animecard-stub
      anime="[object Object]"
    />
    <animecard-stub
      anime="[object Object]"
    />
    <animecard-stub
      anime="[object Object]"
    />
    <!---->
    <!---->
  </v-layout-stub>
</div>
`;
